src/Features/VisualBasic/Portable/InitializeParameter/VisualBasicInitializeMemberFromParameterCodeRefactoringProvider.vb

```diff
@@ -40,5 +41,27 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.InitializeParameter
         Protected Overrides Sub InsertStatement(editor As SyntaxEditor, functionDeclaration As SyntaxNode, method As IMethodSymbol, statementToAddAfterOpt As SyntaxNode, statement As StatementSyntax)
             InitializeParameterHelpers.InsertStatement(editor, functionDeclaration, statementToAddAfterOpt, statement)
         End Sub
+
+        Protected Overrides Function DetermineDefaultFieldAccessibility(document As Document, functionDeclaration As SyntaxNode, model As SemanticModel, cancellationToken As CancellationToken) As Accessibility
+            Dim accessibilityLevel = Accessibility.[Public]
+            Dim service = document.GetLanguageService(Of ISyntaxFactsService)()
+            Dim containingTypeNode = service.GetContainingTypeDeclaration(functionDeclaration, functionDeclaration.SpanStart)
+
+            If containingTypeNode IsNot Nothing Then
+                Dim containingTypeSymbol = CType(model.GetDeclaredSymbol(containingTypeNode, cancellationToken), INamedTypeSymbol)
+
+                ' Fields are public by default in VB, except in the case of classes and modules.
+                Select Case containingTypeSymbol.TypeKind
+                    Case TypeKind.[Class], TypeKind.[Module]
+                        accessibilityLevel = Accessibility.[Private]
+                End Select
+            End If
+            Return accessibilityLevel
+        End Function
```

Comment:
a simpler way would be to just pass in the INamedTypeSymbol for the containing type from the caller.  then you can just do:

```return If(containingType.TypeKind = ..., Accessibility.Private, Acessibilty.Public)```.